qt/python/mantidqt/utils/writetosignal.py

```diff
@@ -20,9 +17,14 @@ class WriteToSignal(QObject):
     used to transform write requests to
     Qt-signals. Mainly used to communicate
     stdout/stderr across threads"""
+
     def __init__(self, original_out):
         QObject.__init__(self)
-        self.__original_out = original_out
+        # If the file descriptor of the stream is < 0 then we are running in a no-external-console mode
+        if original_out.fileno() < 0:
+            self.__original_out = None
```

Comment:
What do you think about using the Null Object pattern here?

We could have a simple class

```python
class NullOut(object):
    def write(self, _):
        pass
```
and do `self.__original_out = NullOut()` in the case where the file descriptor is `< 0`? It then saves doing to `if` check inside the `write` method each time it is called.

Reply:
I am not sure what we benefit from using a class here - do you think it makes it easier to read? If so then I don't mind changing it, but apart from that using a class would be a slight performance loss (if check versus function call, although the difference would likely be unnoticeable overall).

If you think it is easier to read and understand with a class I will make the change, I don't see much of a difference with either

Reply:
I tend to like the deciding once on what a particular thing means and then deferring to some type to handle that behaviour. In this case we decide once what is going to handle to console io and assign the relevant object.

I would agree it might be overkill in this case as the `Null` class only has 1 behaviour. If we had many places with `if ... is not None` then a Null object would definitely be the way to go. Happy for this to be left as it is at this stage.